### PR TITLE
Vending Machine Overhaul

### LINKED ISSUE #299: liberation stations on centcom require access
In-Game report from cacogen:
Revision info: Server revision 0000000000000000000000000000000000000000 compiled on: 2017-02-20

>Revision (Should be above if you're viewing this from ingame!)

it is not above that is a lie

>General description of the issue

title
>What you expected to happen

the liberation stations on centcom before the rebase didn't require access
>What actually happened

title
>Steps to reproduce if possible

1. have bartender access
2. go to centcom
3. use a liberation station

### PR DESCRIPTION
**Pull request by tortellinitony**
:cl: Tortellini Tony
add: Ported vending machines to tgui.
fix: Removed access restrictions from Liberation Stations.
imgadd: New vending machine icons.
/:cl:

https://vgy.me/upeFhD.png
Fixes #299 
Sprites from @dannno

### CODE REVIEW

**Comment by KittyKathy** [comment]
Can we still get our items super fast or is there a delay between each time it hands out a item? what about that old trick where you attach a wire to a coin and can have a chance for the machine to not eat the coin, is that still in? Looks pretty good.

**Comment by tortellinitony** [comment]
You can still do the coin thing, and yes there is a delay between vending, it should be about a second, maybe less. I don't know what unit timer is in

**Comment by cacogen** [comment]
those look like goon sprites

which isn't a bad thing

**Comment by tortellinitony** [comment]
I took dannno's and added a goon-like roof to them since they looked flat before

File: tgui/src/interfaces/vending.ract [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<ui-display title='Products'>
```
Comment (JohnGinnane) [inline tgui/src/interfaces/vending.ract:1]: This is tgui? Noice

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -336,96 +299,66 @@
...
 
-/obj/machinery/vending/Topic(href, href_list)
+/obj/machinery/vending/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, \
+															datum/tgui/master_ui = null, datum/ui_state/state = physical_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "vending", name, width, height, master_ui, state)
+		ui.open()
+
+/obj/machinery/vending/ui_data()
```
Comment (ghost) [inline code/game/machinery/vending.dm:335]: This proc needs to use the lazy helpers

Please do not initialize lists directly on the object, or in `/new` please use the `LAZY_` list helpers when working with lists.

More about that can be found [here](https://github.com/HippieStation/HippieStation13/blob/master/code/__HELPERS/lists.dm#L459)

and

Please use _istypeless for loops_ you can find more information about this in the [contributing.md](https://github.com/HippieStation/HippieStation13/blob/master/CONTRIBUTING.md)

**Review by ghost — CHANGES_REQUESTED** [review]
Code and Sprites look fine, please make sure you use istypeless loops and use the list helpers 👍

**Review by ghost — APPROVED** [review]

**Pull request merged**